Lib/subprocess.py

```diff
@@ -386,6 +386,14 @@ def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
     If timeout is given, and the process takes too long, a TimeoutExpired
     exception will be raised.
 
+    If the current process recevies a KeyboardInterrupt and
+    cleanup_timeout is non-zero, the child process is given some additional
+    time to finish before it is killed, to allow potential clean-up
+    operations in the child to complete. During this time, a second KeyboardInterrupt
+    will kill the child immediately. If cleanup_timeout is not specified,
+    then any remaining time from the original timeout is used, or 1.0 if no
+    original timeout was specified.
```

Comment:
I dislike a default timeout of 1 second. We should not hardcode arbitrary delay in the subprocess module, it's too hard to guess the "best" value. I prefer to keep the current behaviour if timeout is not set: kill immediately the process.